### PR TITLE
Add PR templating

### LINKED ISSUE #528: Standardize PR messages
Currently, everyone has their own format for PR messages.

I'm proposing a template similarly used by larger projects like [eclipse/che](https://github.com/eclipse/che)

Proposal PR soon

### PR DESCRIPTION
**Pull request by sr229**
### Description

This adds the PR templating system for concise PR messages.

This would be the first PR to use the such format.

### Status
**Finished**; Ready to merge, pending final review.

### Changelog

- add ``.github`` and add the ``PULL_REQUEST_TEMPLATE.md`` file which is the template.

### Issues Related

this resolves #528

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<!-- Please review the following before submitting a PR:
```
Comment (TPGPL) [inline .github/PULL_REQUEST_TEMPLATE.md]: Maybe add a Status header (Ready/WIP)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<!-- Please review the following before submitting a PR:
+osu!'s Article Styling Guide: https://osu.ppy.sh/help/wiki/Article_Style_Guide
+-->
+
+<!-- Please remove the comments once you have formulated your Pull Request Message -->
+
+### What does this PR do?
```
Comment (TPGPL) [inline .github/PULL_REQUEST_TEMPLATE.md]: Into Description?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (sr229) [inline .github/PULL_REQUEST_TEMPLATE.md]: what do you mean by here?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (sr229) [inline .github/PULL_REQUEST_TEMPLATE.md]: which Line?

Comment (TPGPL) [inline .github/PULL_REQUEST_TEMPLATE.md]: Header on L7

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (TPGPL) [inline .github/PULL_REQUEST_TEMPLATE.md]: Add somewhere 
" ### Status
Write status of your pull request: Finished/WIP (Note that your changes should be ready before creating new pull request)"

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (sr229) [inline .github/PULL_REQUEST_TEMPLATE.md]: I don't get what you mean here, please elaborate further

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (sr229) [inline .github/PULL_REQUEST_TEMPLATE.md]: done!

**Comment by TPGPL** [comment]
@ephemeralis com her boi

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+<!-- 
+To make a checklist:
+- [] Entry 1
+- [] Entry 2
+-->
+
+### Related Issues
+
+<!-- If you have any issues related to this PR, link them here.
+ex: this resolves #589
```
Comment (TPGPL) [inline .github/PULL_REQUEST_TEMPLATE.md]: e.g.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+-->
+
+### Checklist <!-- Delete this if you don't have a checklist -->
+<!-- 
+To make a checklist:
+- [] Entry 1
+- [] Entry 2
+-->
+
+### Related Issues
```
Comment (TPGPL) [inline .github/PULL_REQUEST_TEMPLATE.md]: Add a comment about deleting the header if not needed

**Comment by sr229** [comment]
done

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+<!-- 
+To make a checklist:
+- [] Entry 1
+- [] Entry 2
+-->
+
+### Related Issues <!-- Delete this if you don't have any issues related to your PR -->
+
+<!-- If you have any issues related to this PR, link them here.
+ex: this resolves #589
```
Comment (TPGPL) [inline .github/PULL_REQUEST_TEMPLATE.md]: I meant that you should change ex: into e.g.

**Comment by sr229** [comment]
@TPGPL All issues and suggestion applied and fixed!

**Pull request merged**